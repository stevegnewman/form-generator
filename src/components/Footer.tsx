import React from 'react';

export const Footer: React.FC = () => {
  return (
    <footer className="bg-secondary py-4">
      <hr className="border-primary border-t w-1/2 mx-auto mb-4" />
      <div className="text-center text-white text-sm py-10">
        Submitted for review by Steve Newman, 22nd November 2024
      </div>
    </footer>
  );
}; 