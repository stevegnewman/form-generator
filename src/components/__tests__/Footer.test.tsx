import React from 'react';
import { render, screen } from '@testing-library/react';
import { Footer } from '../Footer';

describe('Footer', () => {
  it('renders footer text correctly', () => {
    render(<Footer />);
    expect(screen.getByText(/Submitted for review by steve newman/i)).toBeInTheDocument();
    expect(screen.getByText(/22nd november 2024/i)).toBeInTheDocument();
  });

  it('includes a horizontal rule', () => {
    render(<Footer />);
    expect(screen.getByRole('separator')).toBeInTheDocument();
    expect(screen.getByRole('separator')).toHaveClass('border-primary');
  });

  /* Snapshot tests */
  it('matches snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
}); 