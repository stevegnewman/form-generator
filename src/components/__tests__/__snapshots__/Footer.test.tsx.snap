// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<div>
  <footer
    class="bg-secondary py-4"
  >
    <hr
      class="border-primary border-t w-1/2 mx-auto mb-4"
    />
    <div
      class="text-center text-white text-sm py-10"
    >
      Submitted for review by Steve Newman, 22nd November 2024
    </div>
  </footer>
</div>
`;
